src/classes/RD2_ValidationService.cls

```diff
@@ -90,7 +90,7 @@ public with sharing class RD2_ValidationService {
 
         if (rd.npe03__Contact__c == null) {
             rd.addError(System.Label.RD_ContactMustBelongToAccount);
-        } else if (!extractIds(acct.Contacts).contains(rd.npe03__Contact__c)) {
+        } else if (!UTIL_SObject.extractIds(acct.Contacts).contains(rd.npe03__Contact__c)) {
```

Comment:
@heidmary , similarly here, the common utility method is used. The unchanged behaviour can be verified by creating and enhanced RD where RD donor Contact does not belong to the same household specified as the RD Account.